test/test_cuda.py

```diff
@@ -2075,6 +2075,31 @@ def _worker(t):
             for t in range(num_threads):
                 self.assertEqual(results[t].sum().item(), size * size)
 
+    def testCudaRecordStreamWithWrongDevice(self):
+        t = torch.randn(5, 5)
+        with self.assertRaisesRegex(RuntimeError, "Tensor is not allocated in CUDA"):
+            t.record_stream(torch.cuda.current_stream())
+        # Copy to CUDA
+        ct = t.to("cuda")
+        ct.record_stream(torch.cuda.current_stream())
+
+    @classmethod
+    def _recordStreamFromADifferentProcess(cls, t):
+        inst = cls()
+        # This should throw exception as recrod_stream in a different process will not succeed
+        with inst.assertRaisesRegex(RuntimeError, "Tensor is not allocated by the CUDACachingAllocator instance in current context"):
```

Comment:
This does not test correctly, possibly we created a new TestCase instance here. Currently, when I run 

```
py.test test/test_cuda.py -k testCudaRecordStreamWithinWrongProcess -vs
```

It shows:

```
test/test_cuda.py::TestCuda::testCudaRecordStreamWithinWrongProcess Fail to import hypothesis in common_utils, tests are not derandomized
/data/users/shenli/pytorch/torch/backends/cudnn/__init__.py:107: UserWarning: PyTorch was compiled without cuDNN support. To use cuDNN, rebuild PyTorch making sure the
 library is visible to the build system.    
  "PyTorch was compiled without cuDNN support. To use cuDNN, rebuild "
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/home/shenli/local/miniconda/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap                                                                
    self.run()                      
  File "/home/shenli/local/miniconda/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)                                                                                                                            File "/data/users/shenli/pytorch/test/test_cuda.py", line 2088, in _recordStreamFromADifferentProcess
    inst = cls()                                                        
  File "/data/users/shenli/pytorch/test/common_utils.py", line 592, in __init__
    test_method = getattr(self, method_name)            
AttributeError: 'TestCuda' object has no attribute 'runTest'                                                                                                           
PASSED                                           
```

Even if I replace`t.record_stream(torch.cuda.current_stream())` with a `pass` line, the test still passes with the same output. 

One viable fix is to capture the `RuntimeError` in subprocess, and use mutliprocess queue to pass the message back to the parent process, and then do the assert there.